### PR TITLE
Fix 499 - generating parameter names for external functions

### PR DESCRIPTION
**Pull request by EvanMachusak**
When generating an automatic parameter name for a generic type, we strip the `@` character if it was added normalizing the generic type argument name (e.g., because it's a C# keyword like `decimal`).  Example of a bad name was `nullable_@decimal` which is now produced as `nullable_decimal`.

Also made the `definitions` parameter of `AssemblyCompiler.Compile` no longer optional.  When not specified, the function does nothing.

Tested by CqlToElmTests/FunctionDefinitionTest/External_Function

### CODE REVIEW

File: Cql/CodeGeneration.NET/AssemblyCompiler.cs [hunk 1]
```diff
@@ -71,9 +71,10 @@ public AssemblyCompiler(
 
         public IReadOnlyDictionary<string, AssemblyData> Compile(
             LibrarySet librarySet,
-            DefinitionDictionary<LambdaExpression>? definitions = null)
+            DefinitionDictionary<LambdaExpression> definitions)
```
Comment (ewoutkramer) [inline Cql/CodeGeneration.NET/AssemblyCompiler.cs:74]: This will require the user to create an empty dictionary themselves, why is that necessary?

**Review by ewoutkramer — APPROVED** [review]

File: Cql/CodeGeneration.NET/AssemblyCompiler.cs (hunk 1 above)
Comment (baseTwo) [reply Cql/CodeGeneration.NET/AssemblyCompiler.cs:74]: Resolved, since the usages to this method create them anyway
